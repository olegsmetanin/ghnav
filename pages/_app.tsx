import App, { Container } from 'next/app'

import CssBaseline from '@material-ui/core/CssBaseline'
import Head from 'next/head'
import JssProvider from 'react-jss/lib/JssProvider'
import { MuiThemeProvider } from '@material-ui/core/styles'
import { Provider } from 'react-redux'
import React from 'react'
import { configureStore } from 'modules/main/configureStore'
import getPageContext from 'common/theme/getPageContext'
import withRedux from 'next-redux-wrapper'
import withReduxSaga from 'next-redux-saga'

interface IApp {
  store: any
  pageContext: any
}

class MyApp extends App<IApp> {
  static async getInitialProps({ Component, ctx }) {
    let pageProps = {}

    if (Component.getInitialProps) {
      pageProps = await Component.getInitialProps({ ctx })
    }

    return { pageProps }
  }

  pageContext: any

  constructor(props) {
    super(props)
    this.pageContext = getPageContext()
  }

  componentDidMount() {
    // Remove the server-side injected CSS.
    const jssStyles = document.querySelector('#jss-server-side')
    if (jssStyles && jssStyles.parentNode) {
      jssStyles.parentNode.removeChild(jssStyles)
    }
  }

  render() {
    const { Component, pageProps, store } = this.props
    return (
      <Container>
        <Head>
          <title>My page</title>
        </Head>
        {/* Wrap every page in Jss and Theme providers */}
        <JssProvider registry={this.pageContext.sheetsRegistry} generateClassName={this.pageContext.generateClassName}>
          {/* MuiThemeProvider makes the theme available down the React
              tree thanks to React context. */}
          <MuiThemeProvider theme={this.pageContext.theme} sheetsManager={this.pageContext.sheetsManager}>
            {/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
            <CssBaseline />
            {/* Pass pageContext to the _document though the renderPage enhancer
                to render collected styles on server side. */}
            <Provider store={store}>
              <Component pageContext={this.pageContext} {...pageProps} />
            </Provider>
          </MuiThemeProvider>
        </JssProvider>
      </Container>
    )
  }
}

export default withRedux(configureStore)(withReduxSaga({ async: true })(MyApp))
