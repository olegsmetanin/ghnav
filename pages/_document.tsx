import Document, { Head, Main, NextScript } from 'next/document'

import React from 'react'

// Document component is strongly typed with `@types/next`
export default class MyDocument extends Document {
  render() {
    return (
      <html lang="en" dir="ltr">
        <Head>
          <meta charSet="utf-8" />
          {/* Use minimum-scale=1 to enable GPU rasterization */}
          <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
          <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500" />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </html>
    )
  }
}
