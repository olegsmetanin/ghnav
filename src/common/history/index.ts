import Router from 'next/router'
import { isMatch } from 'lodash'
import { qs2json } from 'common/utils/qs2json'
import { qs2pathname } from 'common/utils/qs2pathname'

export interface IHistoryEntity {
  pathname: string
  query: object
}

export interface IHistory {
  get: () => IHistoryEntity[]
  push: (url) => void
  matchPrevious: (IHistoryEntity) => boolean
  matchPreviousPathname: (IHistoryEntity) => boolean
  isComeBack: boolean
  back: () => void
}

export class History implements IHistory {
  history = []

  router = null

  isComeBack = false

  constructor(router) {
    this.router = router
    if (typeof window !== 'undefined') {
      this.push(window.location.pathname + window.location.search)
    }

    router.events.on('routeChangeComplete', url => {
      this.isComeBack = false
      this.push(url)
    })
  }

  push = url => {
    const pathname = qs2pathname(url)
    const query = qs2json(url)
    const hr = { pathname, query }

    this.history.unshift(hr)
  }

  previous = () => {
    return this.history.length > 0 ? this.history[0] : null
  }

  matchPrevious = linkHref => {
    const previous = this.previous()
    const matched = isMatch(previous, linkHref)
    return matched
  }

  matchPreviousPathname = linkHref => {
    const previous = this.previous()
    const matched = previous && linkHref.pathname === previous.pathname
    return matched
  }

  back = () => {
    this.isComeBack = true
    this.router.back()
  }

  get = () => this.history
}

export const history = new History(Router)
