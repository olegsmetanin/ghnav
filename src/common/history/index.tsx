export * from './history'
export * from './withHistory'
