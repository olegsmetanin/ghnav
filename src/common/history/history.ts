import { ASSET_PREFIX } from 'common/config'
import Router from 'next/router'
import { isMatch } from 'lodash'
import { json2router } from 'common/utils/json2router'
import { qs2json } from 'common/utils/qs2json'
import { qs2pathname } from 'common/utils/qs2pathname'

export interface IHistoryEntity {
  pathname: string
  query: object
}

export interface IHistory {
  get: () => IHistoryEntity[]
  push: (urlLike: any, options?: any) => void
  matchPrevious: (IHistoryEntity) => boolean
  matchPreviousPathname: (IHistoryEntity) => boolean
  back: () => void
}

export class History implements IHistory {
  history = []

  router = null

  constructor(router) {
    this.router = router
    if (typeof window !== 'undefined') {
      const currentPath =
        window.location.pathname.slice(ASSET_PREFIX.length) +
        window.location.search
      this.pushUrl(currentPath)
    }

    router.events.on('routeChangeComplete', url => {
      const currentPath = url.slice(ASSET_PREFIX.length)
      this.pushUrl(currentPath)
    })
  }

  pushUrl = url => {
    const pathname = qs2pathname(url)
    const query = qs2json(url)
    const hr = { pathname, query }

    this.history.unshift(hr)
  }

  push = (urlLike: any, options?: any) => {
    const pathname = urlLike.pathname
    const query = json2router(urlLike.query)
    this.history.unshift(urlLike)

    this.router.push(
      {
        pathname,
        query
      },
      {
        pathname: `${ASSET_PREFIX}${pathname}`,
        query
      },
      options
    )
  }

  previous = () => {
    return this.history.length > 0 ? this.history[0] : null
  }

  matchPrevious = linkHref => {
    const previous = this.previous()
    const matched = isMatch(previous, linkHref)
    return matched
  }

  matchPreviousPathname = linkHref => {
    const previous = this.previous()
    const matched = previous && linkHref.pathname === previous.pathname
    return matched
  }

  back = () => {
    this.router.back()
  }

  get = () => this.history
}

export const history = new History(Router)
