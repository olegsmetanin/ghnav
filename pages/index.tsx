import * as React from 'react'

import { IWithHistoryProps, withHistory } from 'common/history/withHistory'
import { WithRouterProps, withRouter } from 'next/router'
import { WithStyles, createStyles, withStyles } from '@material-ui/core/styles'

import { IIssueListQuery } from 'interfaces/issue'
import { IssueListConnected } from 'modules/issue/containers/IssueListConnected'
import { Layout } from 'common/layout/Layout'
import { MainMenu } from 'common/layout/MainMenu'
import { RepoSelectConnected } from 'modules/repo/containers/RepoSelectConnected'
import Typography from '@material-ui/core/Typography'
import { load } from 'modules/issue/redux/listActions'
import { merge } from 'lodash'
import { qs2json } from 'common/utils/qs2json'

const styles = theme =>
  createStyles({
    root: {
      marginTop: theme.spacing.unit * 14
    }
  })

export interface IIndexProps
  extends Partial<IIssueListQuery>,
    WithStyles<typeof styles>,
    IWithHistoryProps,
    WithRouterProps {}

export class BaseIndex extends React.Component<IIndexProps, {}> {
  static defaultProps = {
    filter: {
      state: 'all'
    }
  }

  static async getInitialProps({ ctx }) {
    const { store, isServer } = ctx

    const qs = ctx.asPath

    const query = merge({}, BaseIndex.defaultProps, qs2json(qs))

    if (isServer) {
      if (query.owner && query.repo) {
        store.dispatch(load(query))
      }
    }

    return query as IIndexProps
  }

  route2query = () => {
    const qs = this.props.router.asPath
    const query = {
      ...BaseIndex.defaultProps,
      ...qs2json(qs)
    }
    return query
  }

  handleOnRepoChange = (repoFullName: string) => {
    const [owner, repo] = repoFullName.split('/')

    const query = {
      ...this.route2query(),
      owner,
      repo
    }

    this.props.history.push(
      {
        pathname: '/',
        query
      },
      {
        shallow: true
      }
    )
  }

  handleOnIssueFilterChange = filter => {
    const query = {
      ...this.route2query(),
      filter
    }

    this.props.history.push(
      {
        pathname: '/',
        query
      },
      {
        shallow: true
      }
    )
  }

  render() {
    const { classes } = this.props

    const query = this.route2query()
    const { owner, repo, filter } = query

    const repoFullName = owner && repo ? owner + '/' + repo : ''
    return (
      <div className={classes.root}>
        <Layout
          title={`Github ${repoFullName ? repoFullName + ' ' : ''}issues`}
        >
          <React.Fragment>
            <MainMenu>
              <Typography variant="h5" color="inherit" noWrap>
                Github issues
              </Typography>
              <RepoSelectConnected
                value={repoFullName}
                onChange={this.handleOnRepoChange}
              />
            </MainMenu>
            <IssueListConnected
              {...{ owner, repo, filter }}
              onFilterChange={this.handleOnIssueFilterChange}
            />
          </React.Fragment>
        </Layout>
      </div>
    )
  }
}

export const Index = withStyles(styles)(BaseIndex)

export const IndexPage = withHistory(withRouter(Index))

export default IndexPage
