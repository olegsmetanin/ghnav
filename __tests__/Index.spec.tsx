import * as React from 'react'

import { Index } from '../pages/index'
/* eslint-env jest */
import IndexPage from '../pages/index'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import createSagaMiddleware from 'redux-saga'
import  { historyStab } from 'common/history/__tests__/history.stab'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { routerStab } from 'common/router/__tests__/router.stab'

const sagaMiddleware = createSagaMiddleware()
const mockStore = configureMockStore([sagaMiddleware])

describe('Index', () => {

  it('renders correct', () => {
    const initialState = {
      issueList: {
        query: {
          filter: {
            state: 'all'
          }
        },
        process: {}
      },
      repoList: {}
    }
    const store = mockStore(initialState)
    const props = {
      owner: 'qwe',
      repo: 'asd',
      filter: {
        state: 'all'
      },
      router: {
        asPath: '/?owner=zeit&repo=next.js',
      },
      history: historyStab()
    }

    const component = renderer.create(
      <Provider store={store}>
        <IndexPage {...props}/>
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('getInitialProps works', async () => {

    const initProps = await Index['getInitialProps']({
      ctx: {
        store: null,
        asPath: '/?owner=zeit&repo=next.js',
        isServer: false
      }
    })

    expect(initProps).toEqual({ filter: { state: 'all' }, owner: 'zeit', repo: 'next.js' })
  })

  it('getInitialProps works on server', async () => {
    const props = {
      ctx: {
        store: {
          dispatch: jest.fn()
        },
        asPath: '/?owner=zeit&repo=next.js',
        isServer: true
      }
    }

    await Index['getInitialProps'](props)

    expect(props.ctx.store.dispatch).toBeCalled()
  })

  it('getInitialProps does not dispatch if owner and repo is undefined ', async () => {
    const props = {
      ctx: {
        store: {
          dispatch: jest.fn()
        },
        asPath: '/',
        isServer: true
      }
    }

    await Index['getInitialProps'](props)

    expect(props.ctx.store.dispatch).toBeCalledTimes(0)
  })

  it('works on change repo', async () => {

    const initialState = {
      issueList: {
        query: {
          owner: 'owner',
          repo: 'repo',
          filter: {
            state: 'all'
          }
        },
        process: {}
      },
      repoList: {
        value: [
          { id: 0, owner: 'owner', repo: 'repo' }
        ]
      },
    }

    const props = {
      owner: 'owner',
      repo: 'repo',
      history: historyStab({
        push: jest.fn()
      }),
      router: routerStab({
        asPath: '/?owner=zeit&repo=next.js',
        pathname: '',
        route: '/',
      })
    }

    const store = configureMockStore()(initialState)
    const tree = mount(
      <Provider store={store}>
        <Index {...props}/>
      </Provider>
    )

    const item = tree.find('input').first()

    item.simulate('change', { target: { value: 'owner/repo' } })

    const menuItem = tree.find('MenuItem').first()

    menuItem.simulate('click')

    expect(props.history.push).toBeCalledWith({"pathname": "/", "query": {"filter": {"state": "all"}, "owner": "owner", "repo": "repo"}}, {"shallow": true})

  })

  it('works with router on filter change', async () => {

    const initialState = {
      issueList: {
        query: {
          owner: 'owner',
          repo: 'repo',
          filter: {
            state: 'all'
          }
        },
        process: {}
      },
      repoList: {}
    }


    const props = {
      owner: 'owner',
      repo: 'repo',
      history: historyStab({
        push: jest.fn()
      }),
      router: routerStab({
        asPath: '/?owner=zeit&repo=next.js',
        pathname: '',
        route: '/',
      })
    }

    const store = configureMockStore()(initialState)
    const tree = mount(
      <Provider store={store}>
        <Index {...props}/>
      </Provider>
    )

    const menu = tree.find('[role="button"]').first()
    expect(menu.exists()).toBe(true)
    menu.simulate('click')

    const menuItem = tree.find('MenuItem').at(2)
    expect(menuItem.exists()).toBe(true)

    menuItem.simulate('click')

    await new Promise(resolve => setImmediate(resolve))

    expect(props.history.push).toBeCalledWith({"pathname": "/", "query": {"filter": {"state": "closed"}, "owner": "zeit", "repo": "next.js"}}, {"shallow": true})
  })

  it('works if repo and owner is undefined', async () => {

    const initialState = {
      issueList: {
        process: {}
      },
      repoList: {}
    }

    const props = {
      owner: 'owner',
      router: {
        asPath: '/',
      },
      history: historyStab({
        push: jest.fn()
      })
    }

    const store = configureMockStore()(initialState)
    const tree = mount(
      <Provider store={store}>
        <IndexPage {...props}/>
      </Provider>
    )

    expect(tree.find('WithStyles(BaseLayout)').props().title).toEqual('Github issues')
  })


})
