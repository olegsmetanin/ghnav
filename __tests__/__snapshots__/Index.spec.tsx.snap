// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correct 1`] = `
<div
  className="BaseIndex-root-1"
>
  <div
    className="BaseLayout-root-2"
  >
    <div
      className="PrimarySearchAppBar-root-3"
    >
      <header
        className="MuiPaper-root-22 MuiPaper-elevation4-28 MuiAppBar-root-13 MuiAppBar-positionFixed-14 MuiAppBar-colorPrimary-20 mui-fixed"
      >
        <div
          className="MuiToolbar-root-49 MuiToolbar-dense-52 MuiToolbar-gutters-50"
        >
          <button
            aria-label="Open drawer"
            className="MuiButtonBase-root-59 MuiIconButton-root-53 MuiIconButton-colorInherit-54 PrimarySearchAppBar-menuButton-5"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-58"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-62"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-268"
            />
          </button>
          <h5
            className="MuiTypography-root-71 MuiTypography-h5-87 MuiTypography-colorInherit-100 MuiTypography-noWrap-97"
          >
            Github issues
          </h5>
          <div
            className="BaseRepoSelect-search-107"
          >
            <div
              className="BaseRepoSelect-searchIcon-108"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-62"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </div>
            <div
              aria-expanded={false}
              aria-haspopup="listbox"
              aria-labelledby="downshift-popper-label"
              aria-owns={null}
              className="BaseRepoSelect-container-106"
              role="combobox"
            >
              <div
                className="MuiFormControl-root-113 MuiFormControl-fullWidth-116"
              >
                <div
                  aria-activedescendant={null}
                  aria-autocomplete="list"
                  aria-controls={null}
                  aria-labelledby="downshift-popper-label"
                  className="MuiInputBase-root-130 MuiInput-root-117 BaseRepoSelect-inputRoot-109 MuiInputBase-fullWidth-139 MuiInput-fullWidth-124 MuiInputBase-formControl-131 MuiInput-formControl-118"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="off"
                    className="MuiInputBase-input-140 MuiInput-input-125 BaseRepoSelect-inputInput-110"
                    disabled={false}
                    id="downshift-popper-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Search repo"
                    required={false}
                    type="text"
                    value="zeit/next.js"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="PrimarySearchAppBar-grow-4"
          />
          <div
            className="PrimarySearchAppBar-sectionDesktop-11"
          >
            <button
              className="MuiButtonBase-root-59 MuiIconButton-root-53 MuiIconButton-colorInherit-54"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-58"
              >
                <span
                  className="MuiBadge-root-147"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-62"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                  <span
                    className="MuiBadge-badge-148 MuiBadge-colorSecondary-150"
                  >
                    4
                  </span>
                </span>
              </span>
              <span
                className="MuiTouchRipple-root-268"
              />
            </button>
            <button
              className="MuiButtonBase-root-59 MuiIconButton-root-53 MuiIconButton-colorInherit-54"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-58"
              >
                <span
                  className="MuiBadge-root-147"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-62"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                    />
                  </svg>
                  <span
                    className="MuiBadge-badge-148 MuiBadge-colorSecondary-150"
                  >
                    17
                  </span>
                </span>
              </span>
              <span
                className="MuiTouchRipple-root-268"
              />
            </button>
            <button
              aria-haspopup="true"
              aria-owns={null}
              className="MuiButtonBase-root-59 MuiIconButton-root-53 MuiIconButton-colorInherit-54"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-58"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-62"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-268"
              />
            </button>
          </div>
          <div
            className="PrimarySearchAppBar-sectionMobile-12"
          >
            <button
              aria-haspopup="true"
              className="MuiButtonBase-root-59 MuiIconButton-root-53 MuiIconButton-colorInherit-54"
              data-test-id="mobile-menu-icon"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-58"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-62"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-268"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
    <div>
      <div
        className="BaseIssueListMenu-root-160"
      >
        <header
          className="MuiPaper-root-22 MuiPaper-elevation4-28 MuiAppBar-root-13 MuiAppBar-positionFixed-14 MuiAppBar-colorDefault-19 mui-fixed BaseIssueListMenu-appBar-161"
        >
          <div
            className="MuiToolbar-root-49 MuiToolbar-dense-52 MuiToolbar-gutters-50"
          >
            <div
              className="BaseIssueListMenu-filter-162"
            >
              <div
                className="BaseIssueListMenu-filterItem-163"
              >
                <h6
                  className="MuiTypography-root-71 MuiTypography-h6-88 MuiTypography-colorInherit-100"
                >
                  State
                </h6>
              </div>
              <div
                className="BaseIssueListMenu-filterItem-163"
              >
                <div
                  className="MuiInputBase-root-130 MuiInput-root-117 MuiInput-underline-121"
                  onClick={[Function]}
                >
                  <div
                    className="MuiSelect-root-164"
                  >
                    <div
                      aria-haspopup="true"
                      aria-pressed="false"
                      className="MuiSelect-select-165 MuiSelect-selectMenu-168 MuiInputBase-input-140 MuiInput-input-125"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      All
                    </div>
                    <input
                      id="state"
                      name="state"
                      type="hidden"
                      value="all"
                    />
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-62 MuiSelect-icon-170"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </header>
      </div>
      <div
        className="MuiGrid-container-171"
      />
    </div>
  </div>
</div>
`;
